src/lib/data_source.py

```diff
@@ -99,6 +100,16 @@ def merge(self, record: Dict[str, Any], aux: Dict[str, DataFrame]) -> Optional[s
         # Merge only needs the metadata auxiliary data table
         metadata = aux["metadata"]
 
+        # If date is provided, make sure it follows ISO format
+        if "date" in record:
+            date = safe_datetime_parse(record["date"], "%Y-%m-%d")
+            if date is None:
+                self.errlog(f"Invalid date:\n{record}")
```

Comment:
I like this idea, but afraid it will make the error log even more verbose. I don't see a good away around this though.

Reply:
If some source is pushing bad dates, we really want to know. There should be zero errors logged by this, otherwise it needs to be fixed.